### DISCUSSION TITLE
Fresh reinstall of PiOS

### DISCUSSION THREAD

**Discussion by rowanbsmith** [hatlabs/discussions#134] [Q&A]
How does one go about formatting and installing a new PiOS on the HatLabs hardware?

I understand to rebuild using the SDCard and on a PC to generate SDCard, but I am not sure how to get a new image onto the NVME, what is the way to do that?

I would like to reinstall the base OS and then add the SHPi software and then the relevant GNSS software for an NTP server.

Thanks

**Comment by mairas** [comment]
The process is actually identical to the SD-card one, but you need an NVMe USB adapter. You can find one at Aliexpress for less than 10 USD: search for "nvme usb adapter" and sort the results by orders. Just make sure it is an NVMe and not a SATA adapter. If you don't want to wait, you can probably find one locally at any store that has a good selection of computing gadgets.

When flashing, just insert the card in the adapter and let it poke out at an angle - it'll work just fine for a temporary job. At the end, however, remember to first unplug the USB cable before removing the NVMe drive. I haven't managed to break anything when doing it accidentally in the wrong order but I don't think the NVMe drives should be hot-plugged.

**Reply by mairas** [reply]
Also, you'll be delighted to see how fast the flashing operation is compared to SD cards... :-)
